matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -574,12 +460,7 @@ internal class DefaultCryptoService @Inject constructor(
             Timber.v("Enabling encryption in $roomId for the first time; invalidating device lists for all users therein")
 
             val userIds = ArrayList(membersId)
-
-            deviceListManager.startTrackingDeviceList(userIds)
-
-            if (!inhibitDeviceQuery) {
-                deviceListManager.refreshOutdatedDeviceLists()
-            }
+            olmMachine!!.updateTrackedUsers(userIds)
```

Comment:
Why unwrap here and not anywhere else? Maybe olmMachine could be defined as lateinit var?

Reply:
If we don't correctly update our tracked users we or the other side might have undecryptable messages thus it's an error to not initialize the `olmMachine` before we hit this part of the code. 

Fetching devices out of the store on the other hand doesn't have this problem and callers handle the case where there isn't a device well enough.

There is a bit of a dissonance here since the previous store could be accessed before you initialized the `CryptoService` while the `OlmMachine` requires a valid `deviceId` to be initialized and the store is abstracted behind the `OlmMachine`. That's why I suggested to move the initialization logic somewhere else than what we currently do, in the sync loop.

It would probably make sense for the `CryptoService` to require the `deviceId` and for it to be initialized as soon as we get a valid device ID from the login and later when we load the session info, wherever that might be.

Reply:
Oh also, no idea about the `lateinit`, I don't really know how that's working.

Reply:
`lateinit` is used to tell that a var will not be null when necessary, but will be not initialized in the constructor, but later. So in this case, if `olmMachine` is declared as lateinit, there will be no need to use `!!`. If used without being initialized, it will still crash of course.

Reply:
Yeah, I think I'm using `lateinit`  nowadays somewhere in the verification service.